buildSrc/src/main/java/org/elasticsearch/gradle/ElasticsearchJavaPlugin.java

```diff
@@ -172,26 +163,26 @@ public static void configureCompile(Project project) {
 
                 compileOptions.setEncoding("UTF-8");
                 compileOptions.setIncremental(true);
-
-                // TODO: use native Gradle support for --release when available (cf. https://github.com/gradle/gradle/issues/2510)
-                final JavaVersion targetCompatibilityVersion = JavaVersion.toVersion(compileTask.getTargetCompatibility());
-                compilerArgs.add("--release");
-                compilerArgs.add(targetCompatibilityVersion.getMajorVersion());
-
+                compileTask.getConventionMapping().map("sourceCompatibility", () -> java.getSourceCompatibility().toString());
```

Comment:
TODO: discuss workaround and fix for the need with gradle core team

Reply:
raised this issue in the gradle slack channel: https://gradle-community.slack.com/archives/CA7UM03V3/p1597391556271600

Reply:
https://github.com/gradle/gradle/issues/14141